src/assets.rs

```diff
@@ -246,17 +261,28 @@ mod tests {
             }
         }
 
-        fn synax_for_file_with_content(&self, file_name: &str, first_line: &str) -> String {
+        fn syntax_for_file_with_content(
+            &self,
+            file_name: &str,
+            first_line: &str,
+            as_stdin: bool,
```

Comment:
If possible, I'm trying to avoid using Booleans as parameters to functions, as Rust does not have named parameters. At the call site
```
self.syntax_for_file_with_content(file_name, "", false)
```
it's not really clear what this `false` stands for. One alternative is to use `enum` with two entries.

For this particular case, I wonder if the overall implementation would look a bit cleaner if we would attach the (optional) user-provided file name to the `InputFile` enum:
``` rust
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum InputFile<'a> {
    StdIn(Option<&'a OsStr>),
    Ordinary(&'a OsStr),
    ThemePreviewFile,
}
```

This way, we wouldn't need the additional `file_name` parameter in `get_syntax`, for example.

Reply:
Yes I did notice not having named parameters nor default ones, I'm glad you brought it up. I didn't realize enum entries can contain tuples! Would it also make sense to have the optional provided file name stored there for Ordinary files?

`#[derive(Debug, Clone, Copy, PartialEq)]
pub enum InputFile<'a> {
    StdIn(Option<&'a OsStr>),
    Ordinary(&'a OsStr, Option<&'a OsStr>),
    ThemePreviewFile,
}`

I started looking into this and I'm thinking it might make sense to evaluate the file-name(s) option(s) at the same time as the files themselves, essentially returning from this function

`fn files(&self) -> Vec<InputFile>`

, with the vector containing the optional file-names there. What do you think?

Reply:
I can't see all the implications of that, but it sounds like an idea worth trying.

`enum`s can contain arbitrary data, not just tuples. If we decide to add more fields to the tuples, we should consider using a `struct`. For example:
``` rust
pub struct OrdinaryFile<'a> {
  filename: &'a OsStr,
  user_provided_name: Option<&'a OsStr>,
}

pub enum InputFile<'a> {
    Ordinary(OrdinaryFile),
    // ...
}
```
(names are just an example)


Note that we are discussing the removal of `InputFile` in other tickets, see #884 and #729.

Reply:
> `enum`s can contain arbitrary data, not just tuples. If we decide to add more fields to the tuples, we should consider using a `struct`.

This sounds like a good idea, I will give it a try.

> Note that we are discussing the removal of `InputFile` in other tickets, see #884 and #729.

Good to know. It seems like keeping the user provided file name (or more generally, name of input) close to the InputFile (or whatever it will become as a result of a redesign) makes sense. I'm expecting the changes in this pull request (including potentially making OrdinaryFile struct) , are still safe to do and hopefully will not overly complicate those tickets.